src/main/java/org/opentripplanner/middleware/models/TripMonitorNotification.java

```diff
@@ -39,22 +42,43 @@ public static TripMonitorNotification createAlertNotification(
         return notification;
     }
 
+    /**
+     * Create a new notification about a change in the trip's arrival or departure time exceeding a threshold.
+     *
+     * @param delayInMinutes The delay in minutes (negative values indicate early times).
+     * @param delayThresholdMinutes The user's threshold for delay notifications in minutes.
+     * @param targetDatetime The actual arrival or departure of the trip
+     * @param delayType Whether the notification is for an arrival or departure delay
+     */
     public static TripMonitorNotification createDelayNotification(
         long delayInMinutes,
-        int delayThreshold,
-        NotificationType delayType)
-    {
+        int delayThresholdMinutes,
+        Date targetDatetime,
+        NotificationType delayType
+    ) {
         TripMonitorNotification notification = new TripMonitorNotification();
         notification.type = delayType;
         if (delayType != NotificationType.ARRIVAL_DELAY && delayType != NotificationType.DEPARTURE_DELAY) {
             LOG.error("Delay notification not permitted for type {}", delayType);
             return null;
         }
+        String delayHumanTime;
+        if (Math.abs(delayInMinutes) <= 1) {
+            delayHumanTime = "about on time";
+        } else if (delayInMinutes > 0) {
+            delayHumanTime = String.format("%d minute%s late", delayInMinutes, delayInMinutes > 1 ? "s" : "");
+        } else {
+            delayHumanTime = String.format("%d minute%s early", delayInMinutes, delayInMinutes < -1 ? "s" : "");
+        }
+
         notification.body = String.format(
-            "The %s time for your itinerary was delayed by %d minutes (your threshold is currently set to %d minutes).",
+            "The %s time for your itinerary is now %s (%s) (your threshold is currently set to %d minutes).",
```

Comment:
I think it might be helpful to include the actual time in this message. Here are some suggestions for how this text could read:

> Your trip is set to arrive on time (9:38am).
> Your trip is set to depart 5 minutes late (8:55am). 
> Your trip is set to arrive 10 minutes late (9:55am).

This might need further wordsmithing though. I'm not sure that we should necessarily be tying the unit tests to these strings at this point. Instead, I would rather have one test verify that the string is rendered correctly, and maybe the rest check that the notification gets sent and what the delay value is.

Also, I don't think we should necessarily be telling the user about their threshold -- I don't think that will make any sense in the context of an SMS message.

Reply:
The tests developed already check all these things at once and also for when notifications shouldn't be sent. There might be some value into breaking up the tests into more fine-grained unit tests to be able to test things in isolation, but I'm inclined to keep the test structure as is since it gets a lot of bang-for-the-buck by checking the exact string because in doing so it checks other timing aspects.

I did go ahead and change the message as you recommended in a8d90ab7a8ea11d1ed9c3c64e90e122e4a8a6d14